src/client/Microsoft.Identity.Client/Internal/Requests/LegacyMsiAuthRequest.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.ManagedIdentity;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Internal.Requests
+{
+    /// <summary>
+    /// Old implementation of MSI, which works by getting tokens directly from the MSI infrastructure. 
+    /// </summary>
+    internal class LegacyMsiAuthRequest
```

Comment:
I would replace usages o "Msi" in naming (classes, methods) with "ManagedIdentity" for consistency with the current naming.